plugin/src/main/kotlin/net/octyl/ts2kt/gradle/repository/npm/NpmClientRepository.kt

```diff
@@ -141,15 +157,28 @@ class NpmClientRepository(private val project: Project) : ClientRepository {
 
         // copy to temporary file first
         val temporaryFile = downloadTarget.resolveSibling(".dl.${downloadTarget.name}")
-        val fileChannel = Files.newByteChannel(temporaryFile.toPath(),
-                StandardOpenOption.WRITE,
-                StandardOpenOption.CREATE,
-                StandardOpenOption.TRUNCATE_EXISTING)
+        // hold on to channel to close later
+        val writeChannel = temporaryFile.writeChannel()
         // default arguments don't work here due to KT-24461
-        call.response.content.copyTo(fileChannel)
+        call.response.content.copyTo(writeChannel, Long.MAX_VALUE)
+        // close write channel for safe rename
+        writeChannel.flush()
+        writeChannel.close()
+      
         // then do a move to the actual file, which is less likely to be interrupted
-        temporaryFile.renameTo(downloadTarget)
+        try {
+            // Platform-independent rename/move operation
+            Files.move(temporaryFile.toPath(), downloadTarget.toPath(), StandardCopyOption.REPLACE_EXISTING)
+        } catch (t: Throwable) {
+            // More descriptive file rename error information
+            return ResolutionResult.Error(t)
+        }
 
+        /* Alternatively, use the old operation with less error feedback. I wasn't able to get this working on Windows. /timothyolt
```

Comment:
I would remove this comment, the `Path` APIs are much better and there's no reason we'd revert back.

Reply:
Will do, wrote this a couple days ago, put it in a comment because the PR didn't exist.